### PR TITLE
obtaining timeRangeSelect options dynamically

### PR DESCRIPTION
**Pull request by didof**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area tests

**What this PR does / why we need it**:

These modifications allow the dynamic obtaining of the list of options to be shown in the `TimeRangeSelect` component.
The values, rather than being hard-coded, are generated based on a `builderConfig` object. The function to which it is provided takes care of the construction of both the value and the appropriate label.

**Which issue(s) this PR fixes**:

The insertion/removal of new options is as immediate as adding/removing an integer from the appropriate property in the `builderConfig` object.

Fixes #

**Special notes for your reviewer**:

* The changes allow the addition of options that refer to the seconds.

### CODE REVIEW

File: frontend/src/components/TimeRangeSelect.vue [hunk 1]
```diff
@@ -1,58 +1,80 @@
...
+      clearable
+      v-bind="$attrs"
+      v-on="$listeners"
     />
 </template>
 
 <script lang="ts">
 import Vue from 'vue'
 
+interface BuilderConfig {
```
Comment (fjogeleit) [inline frontend/src/components/TimeRangeSelect.vue:19]: I think you can improve this by changing it to:

```ts
interface BuilderConfig {
  [key in BuilderConfigKeys]: number[];
}
```

if so you should be able to remove the `as BuilderConfigKeys[]` in your method. I know this works for enums, not sure if also by string literals. I'm using enums more often.

File: frontend/src/components/TimeRangeSelect.vue [hunk 2]
```diff
@@ -1,58 +1,80 @@
...
+  },
+  methods: {
+    buildTimeFilters (): Option[] {
+      const additional = {
+        minutes: 1,
+        hours: 60,
+        days: 24 * 60
+      }
+
+      const config = this.builderConfig as BuilderConfig
```
Comment (fjogeleit) [inline frontend/src/components/TimeRangeSelect.vue:56]: If you look into the other components I use the generic types for `Vue.extend` to define the interfaces. So you don't need the `as` in your actual code.

```ts
Vue.extend<Data, Methods, Computed, Props>(...)
```

File: frontend/src/components/TimeRangeSelect.vue [hunk 3]
```diff
@@ -1,58 +1,80 @@
...
+
+            return {
+              label,
+              value
+            }
+          })
+
+          accumulator = [...accumulator, ...timeFilters]
+
+          return accumulator
```
Comment (fjogeleit) [inline frontend/src/components/TimeRangeSelect.vue:74]: I would reduce this to

```ts
return [...accumulator, ...timeFilters]
```

**Review by fjogeleit — CHANGES_REQUESTED** [review]
Could you please remove the package-lock.json from your PR? You don't install/change something.

**Comment by Issif** [comment]
@didof what's the status of this PR? We're planning to release v1.0.0 this week and this PR blocks it. Thanks.

File: frontend/src/components/TimeRangeSelect.vue (hunk 1 above)
Comment (didof) [reply frontend/src/components/TimeRangeSelect.vue:19]: I tried as you suggested but I got this error:
```
A computed property name in an interface must refer to an expression whose type is a literal type or a 'unique symbol' type.
A computed property name must be of type 'string', 'number', 'symbol', or 'any'.
```

File: frontend/src/components/TimeRangeSelect.vue (hunk 2 above)
Comment (didof) [reply frontend/src/components/TimeRangeSelect.vue:56]: As suggested, I tried to implement the signature as shown below.

```
interface Data {
  builderConfig: BuilderConfig
  options: Option[]
  }

interface Methods {
  buildTimeFilters(): Option[]
}

interface Options {
  inheritAttrs: boolean;
}

export default Vue.extend<Data, Methods, null, Options>({...
```

However this generates three errors in the template section: 
```
 :items="options"
      v-bind="$attrs"
      v-on="$listeners"
```

for each of the former values I got the error: "Property 'options' does not exist on type 'never'"

**Comment by didof** [comment]
Excuse me, why this got closed?

**Comment by didof** [comment]
I am gonna send it again (adding some of your suggestions).

**Pull request closed without merging**